File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/platform/operations/feature/FeatureServiceOperationExecutor.java
package org.osgeo.mapguide.test.platform.operations.feature;

import org.osgeo.mapguide.*;
import org.osgeo.mapguide.test.*;
import org.osgeo.mapguide.test.common.*;

import com.almworks.sqlite4java.*;

public abstract class FeatureServiceOperationExecutor extends PlatformApiTestExecutor
{
    protected MgFeatureService _featureService;

    protected FeatureServiceOperationExecutor(String opName, MgFeatureService featSvc, String dbPath)
    {
        super(opName, ApiTypes.Platform, dbPath);
        _featureService = featSvc;
    }
}

File: MgDev/UnitTest/WebTier/DotNet/TestCommon/FeatureService/FeatureServiceOperationExecutor.cs
using OSGeo.MapGuide.Test.Common;
using SqliteDotNet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Operations
{
    /// <summary>
    /// The base class of all test executors using the MgFeatureService
    /// </summary>
    /// <typeparam name="T">The test executor subclass type</typeparam>
    public abstract class FeatureServiceOperationExecutor<T> : PlatformApiTestExecutor
    {
        protected MgFeatureService _featureService;

        protected FeatureServiceOperationExecutor(MgFeatureService featSvc, string unitTestVm)
            : base(typeof(T).Name.ToUpper(), ApiTypes.Platform, unitTestVm)
        {
            _featureService = featSvc;
        }
    }
}
